module/zstd/zstd.c

```diff
@@ -560,8 +560,6 @@ zstd_dctx_alloc(void *opaque __maybe_unused, size_t size)
 		 * zstd_free will release this barrier later.
 		 */
 		mutex_enter(&zstd_dctx_fallback.barrier);
-		mutex_exit(&zstd_dctx_fallback.barrier);
```

Comment:
why is it pointless?

Reply:
the enter/exit barrier is not required (see behlendorfs comment). it was my misstake. the single enter is enough. no additional barrier is required. since the single enter is already a barrier

Reply:
ack